src/Migrate/custom/Set-AzMigrateServerReplication.ps1

```diff
@@ -203,6 +286,349 @@ function Set-AzMigrateServerReplication {
                 $ProviderSpecificDetails.TargetAvailabilityZone = $ReplicationMigrationItem.ProviderSpecificDetail.TargetAvailabilityZone
             }
 
+            if ($HasSqlServerLicenseType)
+            {
+                $validLicenseSpellings = @{ 
+                    NoLicenseType = "NoLicenseType";
+                    PAYG = "PAYG";
+                    AHUB = "AHUB"
+                }
+                $SqlServerLicenseType = $validLicenseSpellings[$SqlServerLicenseType]
+                $ProviderSpecificDetails.SqlServerLicenseType = $SqlServerLicenseType
+            }
+            else
+            {
+                $ProviderSpecificDetails.SqlServerLicenseType = $ReplicationMigrationItem.ProviderSpecificDetail.SqlServerLicenseType
+            }
+
+            if ($HasUpdateTags -And $HasUpdateTagsOperation)
+            {
+                $IllegalCharKey = New-Object Collections.Generic.List[String]
+                $ExceededLengthKey = New-Object Collections.Generic.List[String]
+                $ExceededLengthValue = New-Object Collections.Generic.List[String]
+
+                foreach ($x in $UpdateTags.Keys)
+                {
+                    if ($x.length -gt 512)
+                    {
+                        $ExceededLengthKey.add($x)
+                    }
+
+                    if ($x -match "[<>%&\?/.]")
+                    {
+                        $IllegalCharKey.add($x)
+                    }
+
+                    if ($($UpdateTags.Item($x)).length -gt 256)
+                    {
+                        $ExceededLengthValue.add($($UpdateTags.Item($x)))
+                    }
+                }
+
+                if ($IllegalCharKey.Count -gt 0)
+                {
+                    throw "InvalidTagNameCharacters : The tag names '$($IllegalCharKey -join ', ')' have reserved characters '<,>,%,&,\,?,/' or control characters. These characters are only allowed for tags that start with the prefix 'hidden, link'."
+                }
+
+                if ($ExceededLengthKey.Count -gt 0)
+                {
+                    throw "InvalidTagName : Tag value too large. Following tag value '$($ExceededLengthKey -join ', ')' exceeded the maximum length. Maximum allowed length for tag value - '512' characters."
+                }
+
+                if ($ExceededLengthValue.Count -gt 0)
+                {
+                    throw "InvalidTagValueLength : Tag value too large. Following tag value '$($ExceededLengthValue -join ', ')' exceeded the maximum length. Maximum allowed length for tag value - '256' characters."
+                }
+
+                if ($UpdateTagsOperation -eq "Merge")
```

Comment:
check this implementation with Deepika and here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/tag-resources

Reply:
https://stackoverflow.com/questions/25123877/is-there-a-way-to-make-certain-functions-private-in-a-powershell-script/25129094

Reply:
Explore if we can reduce lines of code by having a dictionary (VmTag: $VmTag, ...) and iterate through it, or maybe private functions that don't get exposed.

Reply:
Done.